TCD-2913: Backtrace - last set of pages (#2292)
saucelabs/sauce-docs: docs/error-reporting/advanced/bcd.md

BCD allocates memory in a separate process on errors, which may not be suitable for systems that lack overcommit.

### Design

During `bcd_init`, BCD initializes a pipe and forks a process. The child process initializes a UNIX socket for per-thread communications. The pipe is used early in initialization to communicate configuration errors and for fatal error handling. The BCD child process forks and executes in response to trace requests.

All communication between the host application and BCD occurs synchronously to maintain global ordering of all BCD operations.

### Handling Crashes

By default, BCD does not set any signal handlers as complex runtimes may rely on signals for functionality. To handle crashes, ensure you install a signal handler. You can use `bcd_emit` for recoverable crashes and `bcd_fatal` for non-recoverable crashes. These functions are signal-safe and thread-safe.

> Here is a simple example that uses `signal`. For production use, it is recommended to use `sigaction` with `SA_SIGINFO` set to allow for additional data extraction at the time of the fault.

Author: ```suggestion
Here is an example that uses `signal`. For production use, it is recommended to use `sigaction` with `SA_SIGINFO` set to allow for additional data extraction at the time of the fault.
```
